packages/email-plugin/src/template-loader.ts

```diff
@@ -1,20 +1,32 @@
-import { LanguageCode } from '@vendure/common/lib/generated-types';
-import fs from 'fs-extra';
+import { Injector, RequestContext } from '@vendure/core';
+import fs from 'fs/promises';
 import path from 'path';
+import { LoadTemplateInput, Partial, TemplateLoader } from './types';
 
 /**
  * Loads email templates according to the configured TemplateConfig values.
  */
-export class TemplateLoader {
-    constructor(private templatePath: string) {}
+export class FileBasedTemplateLoader implements TemplateLoader {
+
+    constructor(private templatePath: string) { }
 
     async loadTemplate(
-        type: string,
-        templateFileName: string,
+        _injector: Injector,
+        _ctx: RequestContext,
+        { type, templateName }: LoadTemplateInput,
     ): Promise<string> {
-        // TODO: logic to select other files based on channel / language
-        const templatePath = path.join(this.templatePath, type, templateFileName);
-        const body = await fs.readFile(templatePath, 'utf-8');
-        return body;
+        const templatePath = path.join(this.templatePath, type, templateName);
+        return fs.readFile(templatePath, 'utf-8');
+    }
+
+    async loadPartials(): Promise<Partial[]> {
+        const partialsPath = path.join(this.templatePath, 'partials');
+        const partialsFiles = await fs.readdir(partialsPath);
+        return Promise.all(partialsFiles.map(async (file) => {
+            return {
+                name: path.basename(file, '.hbs'),
+                content: await fs.readFile(path.join(partialsPath, file), 'utf-8')
+            }
+        }));
```

Comment:
This is the default template loader, based on files, i.e. based on how it currently works in 1.9